source/module_elecstate/module_charge/charge_mixing.cpp

```diff
@@ -796,6 +796,13 @@ void Charge_Mixing::mix_rho_real(Charge* chr)
 
 void Charge_Mixing::mix_dmr(elecstate::DensityMatrix<double, double>* DM)
 {
+    if (GlobalV::MIXING_RESTART <= 0 
+        || this->mixing_restart_count <= 0 
+        || !GlobalV::MIXING_DMR )
```

Comment:
Judge GlobalV::MIXING_DMR in function mix_dmr is strange.
I suggest to pass this parameter into this function or judge GlobalV::MIXING_DMR in ESolver.

Reply:
OK, I decide to keep this verbosity in esolver to remind others mixing_dmr is only called in this case.